bin/zkEnv.sh

```diff
@@ -144,9 +144,15 @@ fi
 
 #echo "CLASSPATH=$CLASSPATH"
 
-# default heap for zookeeper server
-ZK_SERVER_HEAP="${ZK_SERVER_HEAP:-1000}"
-export SERVER_JVMFLAGS="-Xmx${ZK_SERVER_HEAP}m $SERVER_JVMFLAGS"
+# ZK_SERVER_MAXRAMPERCENTAGE has higher precedence over ZK_SERVER_HEAP 
+if [ "x$ZK_SERVER_MAXRAMPERCENTAGE" = "x" ]
+then
+  # default heap for zookeeper server
+  ZK_SERVER_HEAP="${ZK_SERVER_HEAP:-1000}"
+  export SERVER_JVMFLAGS="-Xmx${ZK_SERVER_HEAP}m $SERVER_JVMFLAGS"
+else
+  export SERVER_JVMFLAGS="-XX:MaxRAMPercentage=$ZK_SERVER_MAXRAMPERCENTAGE $SERVER_JVMFLAGS"
+fi
```

Comment:
This seems like it's a bit overly complex. This file is already a config file. It doesn't need additional configuration options to configure the config file.

Users can just edit this file if they need to tweak it. So, it really should just be as simple as possible, and only as complex as necessary for users to be able to easily set the jvm flags.

I suggest something like this:

```suggestion
export SERVER_JVMFLAGS=${SERVER_JVMFLAGS:--Xmx1000m}
```

This gets rid of the unneeded ZK_SERVER_HEAP environment variable and bakes the default into the `SERVER_JVMFLAGS variable`, which you can still override, either by editing this file, or by setting `export SERVER_JVMFLAGS='-XX:MaxRAMPercentage=25'` (or whatever value you want) in your environment. I would suggest a similar thing be done for the `CLIENT_JVMFLAGS`.

Reply:
That makes sense, but how to handle users who are already using ZK_SERVER_HEAP to a value other than 1000? Wouldn't they be forced to change to include -Xmx as part of the SERVER_JVMFLAGS as of now? My concern was that it things are set for them for a long time (by the way of ZK_SERVER_HEAP), suddenly they will be setting their heap back to the default value. How should we handle that?

Reply:
Could do something like:

```bash
if [[ -n $ZK_SERVER_HEAP ]]; then
    echo 'WARNING: ZK_SERVER_HEAP will not be recognized in a future version. Set -Xmx directly using SERVER_JVMFLAGS instead.'
    SERVER_JVMFLAGS="-Xmx${ZK_SERVER_HEAP}m $SERVER_JVMFLAGS"
fi
export SERVER_JVMFLAGS=${SERVER_JVMFLAGS:--Xmx1000m}
```